python_modules/libraries/dagster-dbt/dagster_dbt/core/resources_v2.py

```diff
@@ -1030,7 +1030,13 @@ def _fetch_row_count_metadata(
 
     unique_id = dbt_resource_props["unique_id"]
     logger.debug("Fetching row count for %s", unique_id)
-    table_str = f"{dbt_resource_props['database']}.{dbt_resource_props['schema']}.{dbt_resource_props['name']}"
+    table_str = ".".join(
+        [adapter.quote(part) for part in [
+            dbt_resource_props["database"],
+            dbt_resource_props["schema"],
+            dbt_resource_props["name"]]
+        ]
+    )
```

Comment:
Let's use `relation_name`

```suggestion
    relation_name = dbt_resource_props["relation_name"]
```